pkg/test/deployment.go

```diff
@@ -0,0 +1,78 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/imdario/mergo"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DeploymentOptions struct {
+	Name       string
+	Namespace  string
+	Labels     map[string]string
+	Replicas   int32
+	PodOptions PodOptions
+}
+
+func Deployment(overrides ...DeploymentOptions) *appsv1.Deployment {
```

Comment:
I'm a big fan of simply accepting a deployment object instead of customized options. 
```suggestion
func Deployment(overrides ...appsv1.Deployment) *appsv1.Deployment {
```

Reply:
Not following, it returns a deployment so if you create one to pass it in, there's no point in having this function.

Reply:
This method behaves similarly to `test.Pod`, `test.Node`, `test.DaemonSet`, `test.Provisioner`, `test.PersistentVolume`, etc. It would be the only non-consistent method if it didn't take a `DeploymentOptions`.

Reply:
Not blocking -- just a suggestion